lib/matplotlib/colorbar.py

```diff
@@ -244,6 +244,72 @@ def tick_values(self, vmin, vmax):
         return ticks[(ticks >= vmin) & (ticks <= vmax)]
 
 
+class _ColorbarAutoMinorLocator(ticker.AutoMinorLocator):
+    """
+    AutoMinorLocator for Colorbar
+
+    This locator is just a `.AutoMinorLocator` except the min and max are
+    clipped by the norm's min and max (i.e. vmin/vmax from the
+    image/pcolor/contour object).  This is necessary so that the minorticks
+    don't extrude into the "extend regions".
+    """
+    def __init__(self, colorbar, n=None):
+        """
+        This ticker needs to know the *colorbar* so that it can access
+        its *vmin* and *vmax*.
+        """
+
+        self._colorbar = colorbar
+        self.vmin = self._colorbar.norm.vmin
+        self.vmax = self._colorbar.norm.vmax
+        self.ndivs = n
+
+    def __call__(self):
+        """
+        Return the locations of the ticks. This is a modified version of
+        `class AutoMinorLocator(Locator) from matplot/ticker.py`. Only one
+        line has been changed to use self.vmin and self.vmax for vmin, vmax.
+        """
+        if self.axis.get_scale() == 'log':
+            warnings.warn('AutoMinorLocator does not work with logarithmic '
+                          'scale')
+            return []
+
+        majorlocs = self.axis.get_majorticklocs()
+        try:
+            majorstep = majorlocs[1] - majorlocs[0]
+        except IndexError:
+            # Need at least two major ticks to find minor tick locations
+            # TODO: Figure out a way to still be able to display minor
+            # ticks without two major ticks visible. For now, just display
+            # no ticks at all.
+            return []
+
+        if self.ndivs is None:
+            x = int(np.round(10 ** (np.log10(majorstep) % 1)))
+            if x in [1, 5, 10]:
+                ndivs = 5
+            else:
+                ndivs = 4
+        else:
+            ndivs = self.ndivs
+
+        minorstep = majorstep / ndivs
+        # The following line is the only change made to
+        # the default class AutoMinorLocator(Locator)
+        vmin, vmax = self.vmin, self.vmax
+        if vmin > vmax:
+            vmin, vmax = vmax, vmin
+
+        t0 = majorlocs[0]
+        tmin = ((vmin - t0) // minorstep + 1) * minorstep
+        tmax = ((vmax - t0) // minorstep + 1) * minorstep
+        locs = np.arange(tmin, tmax, minorstep) + t0
+        cond = np.abs((locs - t0) % majorstep) > minorstep / 10.0
+        locs = locs.compress(cond)
+
+        return self.raise_if_exceeds(np.array(locs))
+
 class _ColorbarLogLocator(ticker.LogLocator):
```

Comment:
The CI will likely complain if there are not *2* blank lines before a new class definition.

Reply:
Thanks! just pushed a fix.